src/search/search-index/search.js

```diff
@@ -181,14 +180,25 @@ const boostedUrlSearch = boostedTermSearch(keyGen.url, 0.1)
  * @param {IndexQuery}
  * @returns {Map<string, IndexTermValue>}
  */
-async function bookmarkFilterBackSearch({ limit, skip }) {
+async function bookmarkFilterBackSearch({
+    timeFilter,
+    limit,
+    skip,
+    bookmarksFilter,
+}) {
+    if (bookmarksFilter === false) return null
     const data = []
 
-    data.push(
-        await reverseBookmarks({
-            limit: skip + limit,
-        }),
-    )
+    for (const timeRange of timeFilter.values()) {
+        if (timeRange.gte.split('/')[0] === 'bookmark') {
```

Comment:
I think here it would be better to just get the bookmark time filter object from the `Map` itself, rather than iterating over all values. You should be able to do 
```
const timeRange = timeFilter.get('bookmark/')
data.push(await reverseRangeLookup({ ...timeRange, limit: skip + limit }))
```

(no need for the `for` loop + `if`)